### ISSUE TITLE
AGDPolicy patch not working on my laptop (XPS15 - 9560)

### ISSUE THREAD

**Issue by jardenliu** [acidanthera/bugtracker#283]
HI,
      I   used the latest version of WhateverGreen(v1.2.2). then I found that the HDMI is not output. So i use the debug version release, and I found the AGDPolicy patch did not  enter into force.My laptop is XPS 15 9560( i7-7700HQ), used SMBIOS MacBook Pro 14,3.
     The `isGraphicsPolicyModRequired ` function dont call on my laptop, I dont know why.
      can you tell how to fix?

**Comment by jardenliu** [comment]
my laptop have discrete graphics  which is GTX 1050

**Comment by jardenliu** [comment]
Nvidiagraphicfixup can fix the HDMI issue . why the Whatevergreen cloud not do it？

**Comment by jardenliu** [comment]
yes . I know the optimus technology is not working on mac OS！but the AGDPolicy in Specific SMBIOS will cause HDMI output issue.

**Comment by jardenliu** [comment]
To be honest，I think your understanding of it is flawed.

**Comment by vit9696** [comment]
Okay, I think we get it. So it is not a problem with audio, but with video.
You used an incompatible smbios (MacBookPro14,1 should be the choice for IGPU only), and thus had issues.

The reason why it is not patched for IGPU only configurations by default regardless of you bizarre actions are issues on some desktops. The function is called, and exits immediately due to no discrete GPUs (optimus is not counted): 
https://github.com/acidanthera/WhateverGreen/blob/master/WhateverGreen/kern_weg.cpp#L416. The entry may be missing from macOS logs due to async logging model failure. You can use `liludump=60` to save logs to file.

You can override the agdp mod behaviour by providing `agdpmod=vit9696,pikera` boot argument.

**Comment by jardenliu** [comment]
thanks
